server/src/main/java/org/opensearch/transport/InboundPipeline.java

```diff
@@ -105,6 +105,10 @@ public void close() {
         Releasables.closeWhileHandlingException(decoder, aggregator);
         Releasables.closeWhileHandlingException(pending);
         pending.clear();
+        if (currentHandler != null) {
```

Comment:
@VachaShah This has to go before `Releasables` - otherwise the `currentHandler`, if still executing for whatever reasons, will fail (since everything is released but the handler was not closed)

Reply:
Makes sense! Updated.